codalab/rest/legacy.py

```diff
@@ -145,6 +145,16 @@ def search_worksheets(keywords, worksheet_uuid=None):
         keywords = resolve_owner_in_keywords(keywords)
         results = local.model.search_worksheets(request.user.user_id, keywords)
         BundleService._set_owner_names(results)
+        # Hack: Make sure worksheet infos are JSON serializable.
+        # Since this method is only used for worksheet interpretation (i.e.
+        # resolving wsearch directives), these worksheet infos are never
+        # serialized properly through WorksheetSchema or the Worksheet
+        # ORMObject, and so the `frozen` attribute will cause a serialization
+        # error when we attempt to serialize the interpretation results into
+        # JSON.
+        for info in results:
+            if isinstance(info['frozen'], datetime):
+                info['frozen'] = str(info['frozen'])
```

Comment:
I wouldn't use str()...use isoformat().

I think the `BundleModel` should be responsible for returning things back which are JSON serializable.  This method should not be looking at the internals of `results`.